src/test/java/wooteco/subway/section/SectionServiceTest.java

```diff
@@ -0,0 +1,169 @@
+        SectionDto sectionDto = SectionDto.of(1L, new SectionRequest(2L, 4L, 1));
+        assertThatThrownBy(() -> sectionService.save(sectionDto))
+                .isInstanceOf(SectionInitializationException.class);
+    }
+
+    @Test
+    @DisplayName("역 사이의 거리가 기존 구간의 거리 이상일 경우의 중간 구간 저장")
+    public void saveSectionWithDistanceExceptionCase() {
+        sectionDao.save(1L, 1L, 2L, 10);
+        sectionDao.save(1L, 2L, 3L, 10);
+        SectionDto sectionDto = SectionDto.of(1L, new SectionRequest(2L, 4L, 10));
+        SectionDto sectionDto2 = SectionDto.of(1L, new SectionRequest(2L, 4L, 11));
+
+        assertThatThrownBy(() -> sectionService.save(sectionDto))
+                .isInstanceOf(SectionDistanceException.class);
+        assertThatThrownBy(() -> sectionService.save(sectionDto2))
+                .isInstanceOf(SectionDistanceException.class);
+    }
+
+    @Test
+    @DisplayName("상행 종점 구간 등록")
+    public void saveSectionWithUpEndStiationCase() {
+        sectionDao.save(1L, 1L, 2L, 10);
+        SectionDto sectionDto = SectionDto.of(1L, new SectionRequest(4L, 1L, 1));
+        sectionService.save(sectionDto);
+
+        assertThat(sectionDao.findSectionByUpStationId(1L, 4L).get())
+                .usingRecursiveComparison()
+                .isEqualTo(new Section(2L, 1L, fourthStation, firstStation, 1));
+    }
+
+    @Test
+    @DisplayName("하행 종점 구간 등록")
+    public void saveSectionWithDownEndStationCase() {
+        sectionDao.save(1L, 1L, 2L, 10);
+        SectionDto sectionDto = SectionDto.of(1L, new SectionRequest(2L, 4L, 1));
+        sectionService.save(sectionDto);
+
+        assertThat(sectionDao.findSectionByUpStationId(1L, 2L).get())
+                .usingRecursiveComparison()
+                .isEqualTo(new Section(2L, 1L, secondStation, fourthStation, 1));
+    }
+
+    @Test
+    @DisplayName("구간의 양 역이 노선에 모두 포함될 경우의 구간 등록")
+    public void saveSectionWithBothStationContainCase() {
+        sectionDao.save(1L, 1L, 2L, 10);
+        SectionDto sectionDto = SectionDto.of(1L, new SectionRequest(1L, 2L, 1));
+        SectionDto sectionDto2 = SectionDto.of(1L, new SectionRequest(2L, 1L, 1));
+
+        assertThatThrownBy(() -> sectionService.save(sectionDto))
+                .isInstanceOf(SectionInclusionException.class);
+        assertThatThrownBy(() -> sectionService.save(sectionDto2))
+                .isInstanceOf(SectionInclusionException.class);
+    }
+
+    @Test
+    @DisplayName("구간의 양 역이 노선에 아무것도 포함된 것이 없을 경우의 구간 등록")
+    public void saveSectionWithNeitherStationContainCase() {
+        sectionDao.save(1L, 1L, 2L, 10);
+        SectionDto sectionDto = SectionDto.of(1L, new SectionRequest(3L, 4L, 1));
+
+        assertThatThrownBy(() -> sectionService.save(sectionDto))
+                .isInstanceOf(SectionInclusionException.class);
+    }
+
+    @Test
+    @DisplayName("종점 역이 포함된 구간 삭제")
+    public void deleteSectionWithContainingEndStatonCase() {
```

Comment:
```suggestion
    public void deleteSectionWithContainingEndStationCase() {
```

테스트를 아주 꼼꼼하게 작성해 주셨네요!

Reply:
이런거는 어떻게 발견하시는거죠... 매의 눈 인정합니다

Reply:
오타 수정이 안 됐어요!